includes/wp-admin/class-health-check.php

```diff
@@ -306,4 +316,77 @@ public static function debug_information( $info ) {
 
 		return $info;
 	}
+
+	/**
+	 * Threaded Comments tests.
+	 *
+	 * @return array The test result.
+	 */
+	public static function test_threaded_comments() {
+		$result = array(
+			'label'       => \__( 'Threaded (nested) comments enabled', 'activitypub' ),
+			'status'      => 'good',
+			'badge'       => array(
+				'label' => \__( 'ActivityPub', 'activitypub' ),
+				'color' => 'green',
+			),
+			'description' => \sprintf(
+				'<p>%s</p>',
+				\__( 'Your threaded (nested) comments are enabled and working correctly.', 'activitypub' )
+			),
+			'actions'     => '',
+			'test'        => 'test_threaded_comments',
+		);
+
+		if ( '1' !== get_option( 'thread_comments', '0' ) ) {
+			$result['status']         = 'recommended';
+			$result['label']          = \__( 'Threaded (nested) comments are not enabled', 'activitypub' );
+			$result['badge']['color'] = 'orange';
+			$result['description']    = \sprintf(
+				'<p>%s</p>',
+				\__( 'We recommend enabling threaded (nested) comments to improve the user experience and engagement on your site and on the fediverse.', 'activitypub' )
```

Comment:
How does it improve the UX?

Reply:
This is not the final wording (still WIP): If you do not use threaded comments, mastodon can't show the proper conversation thread and even worse, it shows the comments of an author, together with the post as a "thread". Here is some context:

* https://wordpress.org/support/topic/federated-comments-are-shown-in-post-timeline/
* https://wordpress.org/support/topic/wordpress-comments-are-not-always-federated/